sql/moz-fx-data-shared-prod/telemetry_dev_cycle_derived/glean_metrics_stats_v1/query.sql

```diff
@@ -0,0 +1,50 @@
+WITH glean_app_with_parsed_expiry_date AS (
+  SELECT
+    glean.glean_app,
+    glean.metric,
+    glean.type,
+    glean.first_seen_date AS release_date,
+    glean.last_seen_date AS last_date,
+    glean.expires,
+    CASE
+      WHEN glean.expires = "never"
+        THEN NULL
+      WHEN REGEXP_CONTAINS(glean.expires, r'[0-9]{4}-[0-9]{2}-[0-9]{2}')
+        THEN CAST(glean.expires AS DATE)
+      ELSE releases.publish_date
```

Comment:
Should the publish date be set to publish date if no expiry date set? Sounds like potentially this should be a `null` value?

Reply:
null value indicates that it did not expire yet ...

Reply:
so what's happening here is `expires` can either be "never" (-> null) a date (-> the date) or a version number. If it's a version number we take the date of the version being published as the date it expired